addons/iap_mail/static/src/js/services/iap_notification_service.js

```diff
@@ -0,0 +1,49 @@
+/** @odoo-module **/
+
+import { Markup } from 'web.utils';
+import { registry } from "@web/core/registry";
+
+export const iapNotificationService = {
+    dependencies: ["notification"],
+
+    start(env, { notification }) {
+        env.bus.on("WEB_CLIENT_READY", null, async () => {
+            const legacyEnv = owl.Component.env;
+            legacyEnv.services.bus_service.onNotification(this, (notifications) => {
+                for (const { payload, type } of notifications) {
+                    if (type === 'iap_notification') {
+                        if (payload.error_type == 'success') {
+                            displaySuccessIapNotification(payload);
+                        } else if (payload.error_type == 'danger') {
+                            displayFailureIapNotification(payload);
+                        }
+                    }
+                }
+            });
+            legacyEnv.services.bus_service.startPolling();
+        });
+
+        /**
+         * Displays the IAP success notification on user's screen
+         */
+        function displaySuccessIapNotification(notif) {
+            notification.add(notif.title, {
+                type: notif.error_type,
+            });
+        }
+
+        /**
+         * Displays the IAP failure notification on user's screen
+         */
+        function displayFailureIapNotification(notif) {
+            const message = Markup`<a class='btn btn-link' href='${escape(notif.url)}' target='_blank' ><i class='fa fa-arrow-right'></i> ${_.escape(env._t("Buy more credits"))}</a>`;
```

Comment:
The entire point of `Markup` as template tag is it automatically escapes what's injected (not to be confused with calling `Markup()` on a template string, which does not do that).

So the explicit `_.escape` calls are unnecessary.

Reply:
@xmo-odoo I have removed both _.escape and escape. Thanks

cc @arj-odoo @mba-odoo